### ISSUE TITLE
Replacing Page component in embedded apps

### ISSUE THREAD

**Issue by adriana45silva** [Shopify/polaris-react#1902]
## Issue summary

[Page](https://polaris.shopify.com/components/structure/page) component is used for wrapping other components and actions in a page. Its usage is being deprecated inside of embedded apps, moving the actions `props` to the [Titlebar](https://help.shopify.com/en/api/embedded-apps/app-bridge/react-components/titlebar) component. 
The suggestion made [in this thread](https://shopify.slack.com/archives/C4Y8N30KD/p1564492869056800) was to use the `Layout` component instead of the `Page` with the deprecation rationale. However, when switching its usage for the `Layout` component, the results aren't the same.

Is there other component that we should use instead of the `Page`? Should we continue to work with `Page` component even though the `title` prop is required? 

## Expected behavior

Switching `Layout` component for `Page` component for outer wrapping in pages doesn't give the same layouts.

<img width="1080" alt="Screenshot 2019-07-31 at 11 41 55" src="https://user-images.githubusercontent.com/7239668/62201953-a82daa80-b388-11e9-9071-4291d8559341.png">

## Actual behavior

<img width="1284" alt="Screenshot 2019-07-31 at 11 42 41" src="https://user-images.githubusercontent.com/7239668/62201925-9e0bac00-b388-11e9-923a-133d20ff147a.png">

## Steps to reproduce the problem

You could check the behavior in [this codesandbox](https://codesandbox.io/s/jolly-mahavira-gy0e5?fontsize=14) by replacing `Layout` by `Page`

## Specifications

- Are you using the React components? (Y/N): Yes
- Polaris version number: 3.15.0
- Browser: Chrome 75.0.3770.142
- Operating System: mac Mojave 10.14.3

**Comment by markiiikram** [comment]
Hi @adriana45silva, Thanks for creating this issue. My next few days are booked up, and Monday is a holiday in Canada, but I could look at this issue next Tuesday. Does that work for you?

**Comment by jeffblake** [comment]
My understanding of that deprecation is that it only refers to using certain props on the `Page` such as `breadcrumbs` to render in the embedded app breadcrumb. A `TitleBar` should be used in conjunction with a `Page` in an embedded app. I hope that's the case, the `Page` component looks very good in an embedded app and I've been using it on, well, every page.

**Comment by adriana45silva** [comment]
@jeffblake Hey there! 👋Yes, I'm still using the `Page` component in all my pages and moving all the actions (breadcrumbs, primaryActions, so on) into the `TitleBar` component, but I just wanted to understand if we'll be able to continue using it (or what should we replace the component for in embedded apps). The only caveat, though, is that the `title` prop is required on `Page` component, so it seems weird to pass an empty string there as a value 
(`<Page title="">... </Page>`) since I can't pass the same values there (or everything will be rendered twice and "break" the layout). Maybe making `title` optional could be a feature to be added later on... 🤔

**Comment by markiiikram** [comment]
Hey @adriana45silva, totally, that's what I'd like to look into and if possible work with the App Bridge team to solution.

**Comment by janklimo** [comment]
Hi @markiiikram 👋 checking in to see if there's been progress on this. I'm using `Page` to set titles but also to organize my layout. Moving the title-setting part to `TitleBar` makes a lot of sense, but I think we still need a component that would visually do what `Page` is doing right now (most notably it adds `max-width` and `padding`).

**Comment by markiiikram** [comment]
Hey @janklimo, Apologies for the delay. @iainmcampbell from App Bridge is going to take a look. Thanks Iain!

**Comment by iainmcampbell** [comment]
I’ve done some investigation, and I’m comfortable recommending `<Page title="">` as the recommended migration path.

Here’s the legacy method, using Polaris with deprecated embedded app functionality:

```jsx
import { AppProvider, Page } from "@shopify/polaris";

render(
  <AppProvider apiKey="apikey" shopOrigin="myshop.myshopify.com">
    <Page
      title="My App"
      breadcrumbs={[{ content: "Breadcrumb" }]}
      primaryAction={primaryAction}
      secondaryActions={secondaryActions}
      actionGroups={actionGroups}
    >
      {content}
    </Page>
  </AppProvider>
);

```

which renders:

```jsx
<div class="Polaris-Page">
  <div class="Polaris-Page__Content">
    {...content}
  </div>
</div>
```

Using `<Page title="">` and moving all props previously passed to Page to `<TitleBar>` from App Bridge React:

```jsx
import { AppProvider, Page } from "@shopify/polaris";
import {
  Provider as AppBridgeProvider,
  TitleBar
} from "@shopify/app-bridge-react";

render(
  <AppProvider>
    <AppBridgeProvider
      config={{ apiKey: "apikey", shopOrigin: "myshop.myshopify.com" }}
    >
      <Page title="">
        <TitleBar
          title="My App"
          breadcrumbs={[{ content: "Breadcrumb" }]}
          primaryAction={primaryAction}
          secondaryActions={secondaryActions}
          actionGroups={actionGroups}
        />
        {content}
      </Page>
    </AppBridgeProvider>
  </AppProvider>
);
```

Passing an empty string to `<Page>` does feel a bit wrong, but in this situation the Page component actually does exactly what we want it to. There’s [logic in `<Page>` to ensure we only render the header if header content exists](https://github.com/Shopify/polaris-react/blob/df56f15e238f45879be880fbc5acdbec95cbd0d5/src/components/Page/Page.tsx#L125), so we render exactly the same markup as the legacy method:

```jsx
<div class="Polaris-Page">
  <div class="Polaris-Page__Content">
    {...content}
  </div>
</div>
```

I’ll update the [TitleBar help page](https://help.shopify.com/en/api/embedded-apps/app-bridge/react-components/titlebar) with this recommended migration path.

*Note: in my testing, `<Page></Page>` with no `title` attribute actually didn’t throw any warnings or errors in my editor (vscode) or during compilation. This is odd, as [`title` is not an optional prop](https://github.com/Shopify/polaris-react/blob/df56f15e238f45879be880fbc5acdbec95cbd0d5/src/components/Page/components/Header/components/Title/Title.tsx#L10-L11), so I’m attributing it to a config error on my part.*

**Comment by iainmcampbell** [comment]
Update: `title` will be an optional prop in the next release of Polaris (https://github.com/Shopify/polaris-react/pull/2082).

**Comment by iainmcampbell** [comment]
@janklimo @jeffblake @adriana45silva does this ⬆️ address the issue for you?

**Comment by janklimo** [comment]
@iainmcampbell that works well. One solid benefit I see from decoupling `Page` and `TitleBar` is that now you can have just one `Page` per, well, page. To give a simplified example:

```typescript
const Dashboard: FunctionComponent<Props> = () => {
  const [currentTab, setCurrentTab] = useState<number>(0);

  const renderActiveTab = () => {
    switch (currentTab) {
      case 0:
        return <Galleries />;
      case 1:
        return <Settings />;
      case 2:
        return <Help />;
    }
  };

  return (
    <>
      <Page title="">
        <Tabs tabs={tabs} selected={currentTab} onSelect={setCurrentTab} />
      </Page>
      {renderActiveTab()}
    </>
  );
};
```

Here I use the first `Page` component to add styles to the tabs section. I have to set title to empty and render active tab outside of this component because all of them specify title.

With this change I can see this cleans it up nicely: everything wrapped in `Page`, every tab declaring a `TitleBar`.

**Comment by iainmcampbell** [comment]
This was shipped in [`4.2.0`](https://github.com/Shopify/polaris-react/blob/master/CHANGELOG.md#420---2019-09-09) and a docs update will ship shortly.
